src/helpers/utilities.ts

```diff
@@ -408,3 +412,38 @@ export const fromWei = (number: BigNumberish): string =>
 export const delay = (ms: number): Promise<void> => {
   return new Promise(resolve => setTimeout(resolve, ms));
 };
+
+/**
+ * Creates an object composed of the picked object properties.
+ * @param obj The source object
+ * @param paths The property paths to pick
+ */
+export const pickFlatten = <T, K extends keyof T>(
+  obj: T,
+  paths: K[]
+): Pick<T, K> => {
+  return paths.reduce((acc, key) => {
+    if (typeof obj[key] !== 'undefined') {
```

Comment:
```suggestion
    if (obj[key] !== undefined) {
```

I know you and @terrysahaidak like this pattern, but i see no reason why so use this here

Reply:
I use `typeof`, because if we pass in the path prop which does not exist in the object it will not return true for undefined. Like `obj['noExist'] === 'undefined' //false` `typeof obj['noExist'] === 'undefined' //true`

Reply:
no, this is not true
<img width="311" alt="image" src="https://user-images.githubusercontent.com/25709300/172593207-6d34501e-70fc-4e76-8aed-f99299ad4339.png">